Library/ApplePayCheckoutViewModel.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+import KsApi
+import Prelude
+import ReactiveSwift
+
+public protocol ApplePayCheckoutViewModelInputs {
+  func applePayButtonTapped()
+  func applePayContextDidCreatePayment(with paymentMethodId: String)
+  func applePayContextDidComplete()
+}
+
+public protocol ApplePayCheckoutViewModelOutputs {
+  var goToApplePayPaymentAuthorization: Signal<PostCampaignPaymentAuthorizationData, Never> { get }
+  var completeCheckoutWithApplePayInput: Signal<GraphAPI.CompleteOnSessionCheckoutInput, Never> { get }
+  var validateCheckoutError: Signal<ErrorEnvelope, Never> { get }
+  var createNewPaymentIntentError: Signal<ErrorEnvelope, Never> { get }
+}
+
+public class ApplePayCheckoutViewModel: ApplePayCheckoutViewModelInputs, ApplePayCheckoutViewModelOutputs {
+  // MARK: Late Pledge Init
+
+  /// Crowdfunding logic may need to be in it's own init.
+
+  /*
+
+   Core Late Pledge Functionality:
+   1) Apple pay button tapped
+   2) Create a new payment intent
+   3) Present the payment authorization form
+   4) Payment authorization form calls applePayContextDidCreatePayment with the payment method Id
+   5) Payment authorization form calls paymentAuthorizationDidFinish
+   6) Validate checkout using the checkoutId, payment source id, and payment intent
+   7) Notify the parent View Model that Apple Pay checkout is complete
+   */
+
+  public init(
+    withConfigurationSignal configureWithData: Signal<PostCampaignCheckoutData, Never>,
```

Comment:
Have you explored just giving the ApplePayViewModel the parameters it actually needs, instead of passing along the entire `PostCampaignCheckoutData`? I think just using the parameters it needs would both be clearer and it would hopefully allow us to just use it in the crowdfunding checkout.